lib/msf/base/sessions/meterpreter.rb

```diff
@@ -180,32 +180,23 @@ def bootstrap(datastore = {}, handler = nil)
       print_warning('Meterpreter start up operations have been aborted. Use the session at your own risk.')
       return nil
     end
-    # Unhook the process prior to loading stdapi to reduce logging/inspection by any AV/PSP
-    if datastore['AutoUnhookProcess'] == true
-      console.run_single('load unhook')
-      console.run_single('unhook_pe')
-    end
-
-    unless datastore['AutoLoadStdapi'] == false
-
-      session.load_stdapi
-
-      unless datastore['AutoSystemInfo'] == false
-        session.load_session_info
-      end
-
-      # only load priv on native windows
-      # TODO: abstract this too, to remove windows stuff
-      if session.platform == 'windows' && [ARCH_X86, ARCH_X64].include?(session.arch)
-        session.load_priv rescue nil
-      end
-    end
-
-    # TODO: abstract this a little, perhaps a "post load" function that removes
-    # platform-specific stuff?
-    if session.platform == 'android'
-      session.load_android
+    extensions = datastore['AutoLoadExtensions']&.split(';') || []
+
+    # BEGIN: This should be removed on MSF 7
+    extensions.push('unhook') if datastore['AutoUnhookProcess'] && session.platform == 'windows'
+    extensions.push('stdapi') if datastore['AutoLoadStdapi']
+    extensions.push('priv') if datastore['AutoLoadStdapi'] && session.platform('windows')
```

Comment:
is it `session.platform('windows')` or `session.platform == 'windows' ? I would be nice to pick one and stick to it.